helpers/consumer.go

```diff
@@ -0,0 +1,43 @@
+package helpers
+
+import (
+    "sync"
+)
+
+type Consumer interface {
+    run()
+}
+
+type consumerImpl struct {
+    queue          *chan TransferOperation
+    wg             *sync.WaitGroup
+    maxConcurrency int
+}
+
+func newConsumer(queue *chan TransferOperation, wg *sync.WaitGroup, maxConcurrency int) Consumer {
+    return &consumerImpl{
```

Comment:
Can we make maxConcurrency more closely match the use case, something like maxConcurrentTransfers or maxConcurrentOperations?